pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -470,9 +490,7 @@ def check_monitor_top_k(self, trainer, current: Optional[torch.Tensor] = None) -
     def _format_checkpoint_name(
         cls,
         filename: Optional[str],
-        epoch: int,
-        step: int,
-        metrics: Dict[str, Any],
+        metrics: Dict[str, _METRIC],
```

Comment:
if `metrics` is read-only, we can type with `Mapping` to indicate this

Reply:
It is no longer modified after the call to `_monitor_candidates()`.
So we could type it after that has been called, but it's not a hard requirement.

Do you want me to change it?

Reply:
I don't have a strong opinion, more of a nice to have

Reply:
I'll save it for when we want to truly enforce immutability